app/controllers/admin/orders_controller.rb

```diff
@@ -5,7 +5,21 @@ class OrdersController < AdminController
     before_action :authenticate_employee!
 
     def index
-      @orders = Order.includes(:employee)
+      @orders = Order.includes(:employee).order(:delivered)
+    end
+
+    def deliver
+      @order = Order.find(params[:id])
+      if @order.delivered
+        redirect_to admin_orders_path, notice: 'Order already delivered'
+      else
+        @order.delivered = true
+        if @order.save
+          redirect_to admin_orders_path, notice: 'Order delivered'
+        else
+          redirect_to admin_orders_path, notice: 'Error'
+        end
+      end
```

Comment:
Do you really need a custom action here? It is not an error, but it is in general better approach to use crud actions instead - I would suggest using `update` here -what do you think?

Reply:
you could create `DeliveriesController` with `update` action